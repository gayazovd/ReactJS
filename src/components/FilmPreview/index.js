import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames/bind';
import { withRouter } from 'react-router-dom';
import s from './style.scss';

const cx = classnames.bind(s);

const FilmPreview = withRouter(props => {
    const { history } = props;
    const { film } = props;
    const { id, title, release_date: releaseDate, poster_path: posterPath, genres } = film;
    return (
        <div
            className={cx('wrapper')}
            onClick={() => {
                history.push(`/detail/${id}`);
            }}
            onKeyDown={() => {}}
            role="presentation"
        >
            <img className={cx('images')} src={posterPath} alt="img" />
            <div className={cx('description')}>
                <h3>{title}</h3>
                <div className={cx('wrapper-genres-date')}>
                    <h5 className={cx('genres')}>{genres[0]}</h5>
                    <div className={cx('date')}>{new Date(releaseDate).getFullYear()}</div>
                </div>
            </div>
        </div>
    );
});
FilmPreview.propTypes = {
    onClick: PropTypes.func.isRequired,
    film: PropTypes.shape({
        title: PropTypes.string.isRequired,
        release_date: PropTypes.string.isRequired,
        poster_path: PropTypes.string.isRequired,
        genres: PropTypes.arrayOf(PropTypes.string).isRequired
    }).isRequired
};

export default FilmPreview;
