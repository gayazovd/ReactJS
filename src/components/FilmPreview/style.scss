.images{
    width: 200px;
    height: 300px;
}
.wrapper{
    display: flex;
    flex-direction: column;
    padding: 10px;
    width: 29%;

}
.description{
    display: flex;
    justify-content: space-between;
    padding: 5px;
}
.date{
    height: 15px;
    border: 2px solid #61616171;
    border-radius: 3px;
  }