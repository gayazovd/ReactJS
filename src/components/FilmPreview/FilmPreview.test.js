import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import FilmPreview from './index';

configure({ adapter: new Adapter() });

describe('FilmPreview component is render', () => {
    const props = {
        film: {
            title: 'TITLE',
            release_date: '2019',
            poster_path: '/sasads/sds',
            genres: ['action']
        }
    };
    const filmpreview = shallow(<FilmPreview {...props} />);
    it('FilmPreview render', () => {
        expect(filmpreview).toMatchSnapshot();
    });
});
