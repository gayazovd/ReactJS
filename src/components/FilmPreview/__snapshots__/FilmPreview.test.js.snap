// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmPreview component is render FilmPreview render 1`] = `
<Route>
  <Component />
</Route>
`;
